analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -321,6 +321,35 @@ class Arithmetic
     int? MultiplicationAndAssignments2(int? i) => (i = null) * (i = 42) * i.Value; // Compliant
 }
 
+class Comparisons
+{
+    void NullIsNeitherSmallerNorBiggerThanANumber(int? i)
+    {
+        i = null;
+        if (i > 0) _ = i.Value;           // Noncompliant, FP: positive, therefore non-empty
+
+        i = null;
+        if (i < 0) _ = i.Value;           // Noncompliant, FP: negative, therefore non-empty
+
+        i = null;
+        if (i == 42) _ = i.Value;         // Compliant, 42, therefore non-empty
+
+        i = Unknown();
+        if (i >= 0 || i < 0)
+            _ = i.Value;                  // Compliant, non-empty
+        else
+            _ = i.Value;                  // FN, empty
+
+        i = Unknown();
+        if (i < 0 || i == 0 || i > 0)
```

Comment:
Same as before

Reply:
As in https://github.com/SonarSource/sonar-dotnet/pull/7033#discussion_r1158618441, the SE engine should make the inference: `i ∈ int? && (i < 0 || i == 0 || i > 0) -> i ∈ int? && i ∈ int -> i ∈ int == i.HasValue`.

Reply:
Removed this one too, see https://github.com/SonarSource/sonar-dotnet/pull/7033#discussion_r1159669493